dbtwiz/generate/model.py

```diff
@@ -0,0 +1,156 @@
+import os
+import yaml
+
+from dbtwiz.interact import (
+    input_text,
+    select_from_list,
+    multiselect_from_list,
+    autocomplete_from_list,
+)
+
+from dbtwiz.logging import info, warn, error, fatal
+
+from dbtwiz.model import (
+    Group,
+    Project,
+    layer_choices,
+    materialization_choices,
+    access_choices,
+    domains_for_layer,
+    model_base_path,
+)
+
+
+def generate_model(quick: bool):
+    """Generate new dbt model"""
+
+    project = Project()
+
+    try:
+        layer = select_from_list(
+            "Select model layer",
+            layer_choices())
+
+        domain = autocomplete_from_list(
+            "Which domain does your model belong to",
+            domains_for_layer(layer),
+            must_exist=False,
+            allow_blank=False)
+
+        while True:
+            name = input_text(
+                "What is the name of your model")
+            base_path = model_base_path(layer, domain, name)
+            sql_path = base_path.with_suffix(".sql")
+            yml_path = base_path.with_suffix(".yml")
+            if sql_path.exists() or yml_path.exists():
+                error("A model with that name already exists, please choose another.")
+            else:
+                break
+
+        description = input_text(
+            "Give a short description of your model and its purpose")
+
+        if not quick:
+            group = autocomplete_from_list(
+                "Which group does your model belong to",
+                Group().choices(),
+                must_exist=True,
+                allow_blank=True)
+
+            access = select_from_list(
+                "What is the access level for this model",
```

Comment:
Perhaps rephrase the question to be in the same shape as for materialization:
> "What should the access level be for your model"

Reply:
We should also have a helper text that informs the user when we use the various access levels

Reply:
Perhaps we should then use autocomplete() instead of select()? The alternative is to print all the choices with descriptions above the question, which seems a bit verbose and breaks the style a bit.

Reply:
I don't think that's the case, since this is how it was done in my suggested PR. The description is at the bottom and changes for each:
![image](https://github.com/user-attachments/assets/b2b415b9-0f3b-416d-95dc-bb55bf474231)
![image](https://github.com/user-attachments/assets/6499cbc4-12ff-4e86-b290-851a757adfa4)

Reply:
Refactored the code to add descriptions: https://github.com/amedia/dbtwiz/pull/29/commits/fea4b7e76e24268bde112bec30548518c41d3e7e